templates/definition/meter/smartfox-pro.yaml

```diff
@@ -0,0 +1,113 @@
+  {{- end }}
+  type: custom
+  # jq: parse json generated from response values.xml (https://jqplay.org is your friend to test queries)
+  {{- if not .usage }}
+  power: # heating power in W
+    source: http
+    uri: {{ include "uri" . }}
+    jq: .values.value[] | select(.attrid=="htPowerMeasValue")."#content" | rtrimstr(" kW")
+    scale: 1000
+  {{- end }}
+  {{- if eq .usage "grid" }}
+  power: # grid power in W
+    source: http
+    uri: {{ include "uri" . }}
+    jq: .values.value[] | select(.attrid=="detailsPowerValue")."#content" | rtrimstr(" W")
+  energy: # grid energy in kWh
+    source: http
+    uri: {{ include "uri" . }}
+    jq: .values.value[] | select(.attrid=="energyValue")."#content" | rtrimstr(" kWh")
+  voltages: # grid voltages in V
+  - source: http
+    uri: {{ include "uri" . }}
+    jq: .values.value[] | select(.attrid=="voltageL1Value")."#content" | rtrimstr(" V")
+  - source: http
+    uri: {{ include "uri" . }}
+    jq: .values.value[] | select(.attrid=="voltageL2Value")."#content" | rtrimstr(" V")
+  - source: http
+    uri: {{ include "uri" . }}
+    jq: .values.value[] | select(.attrid=="voltageL3Value")."#content" | rtrimstr(" V")
+  currents: # grid currents in A
+  - source: http
+    uri: {{ include "uri" . }}
+    jq: .values.value[] | select(.attrid=="ampereL1Value")."#content" | rtrimstr(" A")
+  - source: http
+    uri: {{ include "uri" . }}
+    jq: .values.value[] | select(.attrid=="ampereL2Value")."#content" | rtrimstr(" A")
+  - source: http
+    uri: {{ include "uri" . }}
+    jq: .values.value[] | select(.attrid=="ampereL3Value")."#content" | rtrimstr(" A")
+  powers: # grid powers in W
+  - source: http
+    uri: {{ include "uri" . }}
+    jq: .values.value[] | select(.attrid=="powerL1Value")."#content" | rtrimstr(" W")
+  - source: http
+    uri: {{ include "uri" . }}
+    jq: .values.value[] | select(.attrid=="powerL2Value")."#content" | rtrimstr(" W")
+  - source: http
+    uri: {{ include "uri" . }}
+    jq: .values.value[] | select(.attrid=="powerL3Value")."#content" | rtrimstr(" W")
+  {{- end }}
+  {{- if eq .usage "pv" }}
+  power: # PV power in W
+    source: calc
+    add:
+    - source: http
+      uri: {{ include "uri" . }}
+      jq: .values.value[] | select(.attrid=="wr1PowerValue")."#content" | rtrimstr(" kW")
+      scale: 1000
+    - source: http
+      uri: {{ include "uri" . }}
+      jq: .values.value[] | select(.attrid=="wr2PowerValue")."#content" | rtrimstr(" kW")
+      scale: 1000
+    - source: http
+      uri: {{ include "uri" . }}
+      jq: .values.value[] | select(.attrid=="wr3PowerValue")."#content" | rtrimstr(" kW")
+      scale: 1000
+    - source: http
+      uri: {{ include "uri" . }}
+      jq: .values.value[] | select(.attrid=="wr4PowerValue")."#content" | rtrimstr(" kW")
+      scale: 1000
+    - source: http
+      uri: {{ include "uri" . }}
+      jq: .values.value[] | select(.attrid=="wr5PowerValue")."#content" | rtrimstr(" kW")
+      scale: 1000
```

Comment:
Kompakter?

```suggestion
  power: # PV power in W
    source: http
    uri: {{ include "uri" . }}
    jq: [.values.value[]|select(.attrid|test("wr\\d+PowerValue"))."#content"|scan("([-\\+]?\\d+\\.?\\d*)")|.[]|tonumber]|add*1000
```

Reply:
...also die Ursprungsvariante hab ich noch verstanden. Hier würde ich sagen EInfachheit vor clever sein? Oder bitte einen dicken Kommentar ran schreiben was das tut!

Reply:
Sieht man doch ;)
Es summiert die aus den Strings extrahierten Zahlenwerte die zuvor dazu als Zahlentyp umgewandelt und in ein Array geschrieben worden für alle Eintäge in deren Attribut `id` die Suche nach dem String `wr` gefolgt von Zahlen gefolgt von `PowerValue` erfolgreich war.
Ganz einfach :D

Reply:
Das *dachte* ich mir, aber das war auch alles. Ich finde wirklich die Originalversion besser. Die ist super lesbar, wenn auch nicht so fancy.

Reply:
Ich überlasse es euch, was ihr schöner findet. Ein Kommentar wäre bei der kompakten Version aber sicher nicht verkehrt.
Ausserdem würde ich die Skalierung nicht in der jq Query, sondern mit dem Scale ausserhalb machen.

Reply:
@premultiply Leider funktioniert die kompakte Version bei mir nicht. Irgendwie scheint es da noch ein Problem in der jq Expression zu geben.

Ich würde dafür plädieren, die ursprüngliche Version zu behalten. Dieser Teil bezieht sich sowieso nur auf das Auslesen der pv Werte. Und idealerweise bezieht man die lieber direkt vom Wechselrichter als indirekt via Smartfox. Der Smartfox macht eher für grid und aux Sinn.

Reply:
Ich habe das Problem gefunden, die jq Expression für die Summe muss in single quotes eingebettet sein, siehe https://stackoverflow.com/questions/52065570/how-do-i-sum-all-numbers-from-output-of-jq/58233650#58233650

Neuer Commit für kompakte Version mit Kommentar folgt in Kürze...

Reply:
Siehe neue Kompaktversion mit Kommentar. Hoffentlich passt das für euch beide.